cpp/src/arrow/compute/exec/plan_test.cc

```diff
@@ -144,7 +144,14 @@ TEST(ExecPlan, DummyStartProducing) {
   // Note that any correct reverse topological order may do
   ASSERT_THAT(t.started, ElementsAre("sink", "process3", "process2", "process1",
                                      "source2", "source1"));
-  ASSERT_EQ(t.stopped.size(), 0);
+
+  plan->StopProducing();
+  plan->finished().Wait();
+  ASSERT_THAT(t.stopped, ElementsAre("source1", "source2", "process1", "process2",
```

Comment:
Do you need the same comment as above?  Is "source2", "source1", ... valid too?

Reply:
I'll duplicate the comment